apps/ledger-live-desktop/src/main/internal-lifecycle.ts

```diff
@@ -211,29 +211,42 @@ ipcMain.on("setEnv", async (event, env) => {
   }
 });
 
-// Routes a (request) message from the renderer process to the internal process,
-// and sets a handler to receive the response from the internal thread and reply it to the renderer process
-// Only 1 response from the internal process is expected.
+/**
+ * Factory creating a request/response handler on requests from the renderer process that are sent to the internal process.
+ *
+ * The handler routes a (request) message from the renderer process to the internal process,
+ * and sets a handler to receive the response from the internal process and reply it to the renderer process
+ * Only 1 response from the internal process is expected.
+ *
+ * @param channel the channel name to create the handler for
+ */
 const internalHandlerPromise = (channel: string) => {
```

Comment:
I feel like since these are factories (this one, `internalHandlerObservable` and `internalHandlerEvent`) they could benefit from improved naming. (For instance `makeInternalHandlerPromise`).
The impact of renaming affects no other file and I think improves the comprehension for newbies like me reading the code.

Reply:
in case you rename, can you also update the doc please ?  https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/IPCTransport.md#generalisation

Reply:
Yes good idea, and those functions are only used in one file indeed